sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -2244,6 +2244,15 @@ object GpuOverrides {
         ("map", TypeSig.MAP.nested(TypeSig.STRING), TypeSig.MAP.nested(TypeSig.all)),
         ("key", TypeSig.lit(TypeEnum.STRING), TypeSig.all)),
       (in, conf, p, r) => new GpuGetMapValueMeta(in, conf, p, r)),
+    expr[ElementAt](
+      "Returns element of array at given(1-based) index in value if column is array. " +
+        "Returns value for the given key in value if column is map.",
+      ExprChecks.binaryProjectNotLambda(
+        TypeSig.commonCudfTypes, TypeSig.all,
+        ("left", TypeSig.ARRAY.nested(TypeSig.commonCudfTypes) +
+          TypeSig.MAP.nested(TypeSig.STRING), TypeSig.all),
```

Comment:
I tested this case, and got an error like 
```
scala> spark.sql("select element_at(2, 2)").collect
org.apache.spark.sql.AnalysisException: cannot resolve 'element_at(2, 2)' due to data type mismatch: The first argument to function element_at should have been array or map type, but its int type.; line 1 pos 7;
'Project [unresolvedalias(element_at(2, 2), None)]
+- OneRowRelation

  at org.apache.spark.sql.catalyst.analysis.package$AnalysisErrorAt.failAnalysis(package.scala:42)
  at org.apache.spark.sql.catalyst.analysis.CheckAnalysis$$anonfun$$nestedInanonfun$checkAnalysis$1$2.applyOrElse(CheckAnalysis.scala:149)
  at org.apache.spark.sql.catalyst.analysis.CheckAnalysis$$anonfun$$nestedInanonfun$checkAnalysis$1$2.applyOrElse(CheckAnalysis.scala:140)
  at org.apache.spark.sql.catalyst.trees.TreeNode.$anonfun$transformUp$2(TreeNode.scala:333)

```

Generally, I am following existing `GetArrayItem` and `GetMapValue` for this expression.

Reply:
Better to update the type sig for Spark part.
If it is 'all', there will be many unsupported types against Spark in the `supported_ops ` doc. But actually it is not, since Spark does support only map and array.

Reply:
good suggestion! Updated.